reference/vpp/technical-details.md

```diff
@@ -0,0 +1,83 @@
+
+vpp-manager is voluntarily kept as simple as possible, in order to minimize the risk of bugs, as these could leave the host without connectivity, requiring a reboot.
+
+
+#### calico-vpp-agent
+
+The {{ site.prodname }} VPP agent is the process responsible for all the {{ site.prodname }}-specific configuration in VPP. Its code lives in this directory: [https://github.com/projectcalico/vpp-dataplane/tree/master/calico-vpp-agent](https://github.com/projectcalico/vpp-dataplane/tree/master/calico-vpp-agent).
+
+This agent is split in four main components, which interact with the k8s and {{ site.prodname }} APIs to configure VPP. These components are the routing manager, the CNI server, the services manager and the policies manager.
+
+**Routing manager**
+
+The {{ site.prodname }} VPP agent embeds a GoBGP daemon, and dynamically updates its configuration (including peers, ASN, etc.) according to the {{ site.prodname }} configuration. As the calico-vpp-node pod runs in the host network namespace, the BGP daemon uses the host's TCP stack, and sends and receives traffic through the host's interface `vpptap0`.
+
+When routes are added or removed in BGP, the routing manager reflects the changes in VPP. The routes are installed differently depending on the {{ site.prodname }} configuration. If the configuration requires the use of an IPIP or VXLAN tunnel, then the tunnel interface will be created on demand in VPP, and the route will be added through the tunnel. Otherwise, the route is simply added as-is in VPP.
+
+**CNI server**
+
+This component implements a server that receives gRPC request from the {{ site.prodname }} CNI (configured with a gRPC dataplane) through a unix socket mounted on the host.
+
+When it receives an ADD request, the CNI server creates a tap interface in the container's namespace, and configures it with the IP address and routes chosen by {{ site.prodname }}. The routes' next hop is an otherwise unused link-local address, both in IPv4 and in IPv6. A /32 or /128 route is added in VPP as well for the container address through the tap. When it receives a DEL request, the CNI servers cleans up the tap from VPP and from the container's namespace.
+
+**Services manager**
+
+This component is the equivalent of kube-proxy for VPP, i.e. it configures NAT load-balancing rules to implement Kubernetes services in VPP. It watches the Kubernetes Services and Endpoints APIs, and updates the VPP configuration on each change. Service load balancing is implemented with a {{ site.prodname }}-specific DNAT plugin in VPP.
+
+**Policies manager**
+
+This component implements {{ site.prodname }} policies in VPP. Felix ({{ site.prodname }}'s policy agent) is configured to use a lightweight proxy as dataplane. This proxies relays all the configuration messages sent by Felix to the `calico-vpp-agent`, and status updates the other way. The VPP agent then uses a custom plugin in VPP to implement policies.
```

Comment:
to use a lightweight proxy as dataplane. (as a dataplane?)

Reply:
as its dataplane?

Reply:
"as its dataplane" sounds good to me :+1: